installer/datafiles/base_omsagent.data

```diff
@@ -211,6 +224,9 @@ if ${{PERFORMING_UPGRADE_NOT}}; then
    rmdir /etc/opt/microsoft/omsagent 2> /dev/null
    rmdir /etc/opt/microsoft 2> /dev/null
    rmdir /etc/opt 2> /dev/null
+elif ls ${{OMSHOME}}/ruby/lib/ruby/gems/2.2.0/cache/mysql2-0.4.4.gem 1> /dev/null 2> /dev/null; then
```

Comment:
I see. Thanks for pointing to the different rules for how/when postinstall is executed for rpm vs deb based systems. I will pick #1 from your suggested list. It also seems simpler and easy to track since all is done in one place (preinstall).